torchsde/_core/adjoint_sde.py

```diff
@@ -58,32 +58,72 @@ def __init__(self,
             NOISE_TYPES.diagonal: self.gdg_prod_diagonal,
         }.get(sde.noise_type, self.gdg_prod_default)
 
-    def _unpack_y_aug(self, y_aug):
+    def _unpack_y_aug(self, t, y_aug, v):
+        # These leaf checks are very important.
```

Comment:
Could we shorten this comment for a bit? It's becoming design-doc-like. The crux should be that this prevents `grad` from going into `y` towards `params`, right?

Reply:
I'm trying to write for the developer five years from now who's forgotten everything about this. What's the issue with long comments?

Reply:
The reason I'm asking for some simplifications is to prevent the code from getting too bulky to read (I personally had to parse this text for some time!). 

I certainly think this comment section could be shortened a little, but if you insist on including this unmodified, then I'd recommend wrapping this in the docstring, in case we'd like to auto-generate documentation webpages based on docstrings, and that in the rare occasion we'd like to call `help` on this function.

Reply:
I've shortened the comment to the extent that I think's reasonable.
I deliberately didn't put it in the docstring because it's really just about the `.is_leaf` checks rather than the function as a whole.